### ISSUE TITLE
M109 & M190 with no or wrong parameters

### ISSUE THREAD

**Issue by Tannoo** [MarlinFirmware/Marlin#6308]
This causes Marlin to think it's heating, but the heaters do not come on due to no temperature set.
Marlin is then waiting for the temperature to get to an unknown temp without turning on the heaters.
It does timeout, but is it simulating the dryrun that's not enabled?

This needs to error out saying no temp set or something to that effect.

<details><summary>M109 Log Output</summary>

```
echo:M109
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.7 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.5 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.5 /0.0 B:22.4 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.4 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.5 /0.0 B:22.4 /0.0 @:0 B@:0 W:?
 T:18.6 /0.0 B:22.2 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.5 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.3 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.1 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:22.2 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.2 /0.0 @:0 B@:0 W:?
 T:18.2 /0.0 B:22.1 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:22.2 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.1 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.1 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:22.1 /0.0 @:0 B@:0 W:?
 T:18.4 /0.0 B:22.0 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:22.0 /0.0 @:0 B@:0 W:?
 T:18.2 /0.0 B:22.0 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.2 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.2 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.2 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.3 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.2 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.2 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.1 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
 T:18.1 /0.0 B:21.9 /0.0 @:0 B@:0 W:?
ok
```
</details>
<details><summary>M190 Log Output</summary>

```
echo:M190
 T:17.9 /0.0 B:20.2 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.3 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.1 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.2 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.3 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.2 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.1 /0.0 @:0 B@:0 W:?
 T:18.1 /0.0 B:20.1 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.2 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.1 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.1 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.1 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.1 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:18.0 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.9 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.6 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.8 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.7 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
 T:17.5 /0.0 B:20.0 /0.0 @:0 B@:0 W:?
ok
```
</details>

**Comment by thinkyhead** [comment]
Seeing any issues with `M104` and `M140`?

**Comment by thinkyhead** [comment]
I've checked the "specification" — the description on the RepRap wiki — and it says that `M109` and `M190` don't do anything without an `S` or `R` parameter. So I'll just make it bounce if neither of these parameters is included.

Alternatively, without parameters these could just take the current target temperature and go with that, making `M109` / `M190` alone into simple "wait for current target temperature to be reached" commands.

**Comment by Tannoo** [comment]
I would make it bounce.

**Comment by thinkyhead** [comment]
Bounce it is!

**Comment by Tannoo** [comment]
Fixed with PR #6311.
